mlir/tools/mlir-miopen-driver/mlir-miopen-driver.cpp

```diff
@@ -1833,102 +1889,122 @@ static LogicalResult populateValidationLogic(
     // If using random data, emit CPU alloc and copy input data
     if (operation.getValue() == "conv2d" ||
         operation.getValue() == "conv2d_dummy") {
-      cpuFilterHostAllocOp =
+      verifierFilterHostAllocOp =
           allocAndCopyTensor(module, builder, block, mcpuMemCopy5DFuncOp,
                              filterHostAllocOp, filterDimension, convertFuncs);
-      cpuInputHostAllocOp =
+      verifierInputHostAllocOp =
           allocAndCopyTensor(module, builder, block, mcpuMemCopy5DFuncOp,
                              inputHostAllocOp, inputDimension, convertFuncs);
-      cpuOutputHostAllocOp = allocAndInitializeTensor(
+      verifierOutputHostAllocOp = allocAndInitializeTensor(
           builder, block, floatType, mcpuMemset5DFuncOp, outputMemRefType,
           memsetValue, memsetValue, seedConstantIntOp);
     } else if (operation.getValue() == "conv2d_bwd_data") {
-      cpuFilterHostAllocOp =
+      verifierFilterHostAllocOp =
           allocAndCopyTensor(module, builder, block, mcpuMemCopy5DFuncOp,
                              filterHostAllocOp, filterDimension, convertFuncs);
-      cpuInputHostAllocOp = allocAndInitializeTensor(
+      verifierInputHostAllocOp = allocAndInitializeTensor(
           builder, block, floatType, mcpuMemset5DFuncOp, inputMemRefType,
           memsetValue, memsetValue, seedConstantIntOp);
-      cpuOutputHostAllocOp =
+      verifierOutputHostAllocOp =
           allocAndCopyTensor(module, builder, block, mcpuMemCopy5DFuncOp,
                              outputHostAllocOp, outputDimension, convertFuncs);
     } else if (operation.getValue() == "conv2d_bwd_weight") {
-      cpuFilterHostAllocOp = allocAndInitializeTensor(
+      verifierFilterHostAllocOp = allocAndInitializeTensor(
           builder, block, floatType, mcpuMemset5DFuncOp, filterMemRefType,
           memsetValue, memsetValue, seedConstantIntOp);
-      cpuInputHostAllocOp =
+      verifierInputHostAllocOp =
           allocAndCopyTensor(module, builder, block, mcpuMemCopy5DFuncOp,
                              inputHostAllocOp, inputDimension, convertFuncs);
-      cpuOutputHostAllocOp =
+      verifierOutputHostAllocOp =
           allocAndCopyTensor(module, builder, block, mcpuMemCopy5DFuncOp,
                              outputHostAllocOp, outputDimension, convertFuncs);
     }
   }
 
   // Populate host validation logic
-  auto cpuConvFuncOp = createCPUConvolution(module, builder, filterMemRefType,
-                                            inputMemRefType, outputMemRefType);
-
-  // Emit conv2d_host function call.
-  auto cpuConvCallOp = builder.create<CallOp>(
-      builder.getUnknownLoc(), cpuConvFuncOp,
-      ValueRange{cpuFilterHostAllocOp, cpuInputHostAllocOp,
-                 cpuOutputHostAllocOp});
-  block->push_back(cpuConvCallOp);
+  if (populateValidationWithGPU.getValue() &&
+      (xdlopsV2.getValue() || dataType != builder.getF32Type())) {
```

Comment:
Why are we verifying with f32 when the datatype isn't f32 and xdlops are off? In fact, what's this or checking for?

Reply:
We cannot verify non-xdlops with xdlops unless we're on gfx908, so I'm verifying non-XDLops f16/bf16 with non-XDLops f32, no matter we're on gfx908 or not. Non-XDLops f32 is still validated using C++ kernels, as it doesn't make a lot of senses to verify with less precision types.

Reply:
Ok, that makes sense. Would it instead make sense to have this be `&& !(!xdlopsV2.getValue() && dataType == builder.getF32Type())` - while this is logically equivalent to what you have, it might be clearer what you're doing

Reply:
I'm OK with either form.